src/dsp_tools/utils/id_to_iri.py

```diff
@@ -160,6 +160,36 @@ def _replace_ids_by_iris(
     return tree, success
 
 
+def _remove_resources_if_id_in_mapping(
+    tree: etree._Element,
+    mapping: dict[str, str],
+) -> tuple[etree._Element, bool]:
+    """
+    Remove all resources from the XML file if their ID is in the mapping.
+
+    Args:
+        tree: parsed XML file
+        mapping: mapping of internal IDs to IRIs
+
+    Returns:
+        a tuple of the modified XML tree and the success status
+    """
+    success = True
+    resources = tree.xpath("|".join([f"/knora/{x}" for x in ["resource", "annotation", "link", "region"]]))
+    resources_to_remove = [x for x in resources if x.attrib.get("id") in mapping]
+    for resource in resources_to_remove:
```

Comment:
This could also be done in a comprehension (they are faster)

Reply:
A comprehension would indeed work. I would then have 1 line that looks like this: 

```
[resource.getparent().remove(resource) for resource in resources_to_remove]
```

This is an expression that is assigned to nothing, so it invokes this pylint warning: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/expression-not-assigned.html.

Now, I could assign it to `_`:

```
_ = [resource.getparent().remove(resource) for resource in resources_to_remove]
```

That would work for both python and pylint, but I dislike it, because I think it's harder to understand.